cli/src/clients.rs

```diff
@@ -47,6 +47,46 @@ pub async fn create_trunk_repo(
 
     Ok(())
 }
+pub async fn update_bundle_upload_status(
+    origin: &str,
+    api_token: &str,
+    id: &str,
+    upload_status: &str,
+) -> anyhow::Result<()> {
+    let client = reqwest::Client::new();
+    let resp = match client
+        .patch(format!("{}/v1/metrics/updateBundleUpload/{}", origin, id))
+        .timeout(TRUNK_API_TIMEOUT)
+        .header(reqwest::header::CONTENT_TYPE, "application/json")
+        .header(TRUNK_API_TOKEN_HEADER, api_token)
+        .json(&UpdateBundleUploadRequest{
+            upload_status: upload_status.to_owned()
+        })
+        .send()
+        .await
+    {
+        Ok(resp) => resp,
+        Err(e) => return Err(anyhow::anyhow!(e).context("Failed to update bundle upload status")),
+    };
+
+    if resp.status().is_success() {
+        return Ok(());
+    }
+
+    if resp.status().is_client_error() {
+        return Err(anyhow::anyhow!(
+            "Failed to update bundle upload status. Client error: {}",
+            resp.status()
+        ));
+    }
+
+    log::warn!(
+        "Failed to update bundle upload status. {}: {}",
+        resp.status(),
+        status_code_help(resp.status())
+    );
+    Ok(())
```

Comment:
nit: Since this function gets retried it can print `Failed to update bundle upload status...` many times. I see we've already got a log warn happening on the retry, so I think it's okay to remove the `log::warn` here. Separately, I plan to fix up the other API call functions here.

nit: once that's done we can simply the code a bit
```suggestion
    if resp.status().is_client_error() {
        return Err(anyhow::anyhow!(
            "Failed to update bundle upload status. Client error: {}",
            resp.status()
        ));
    }

    Ok(())
```